Services/LaunchServer/Launcher.cpp

```diff
@@ -47,6 +49,33 @@ Launcher& Launcher::the()
     return *s_the;
 }
 
+void Launcher::load_handlers(const String af_dir)
+{
+    auto load_hashtable = [](auto af, auto key) {
+        HashTable<String> table;
+
+        auto config_value = af->read_entry("Launcher", key, {});
+        for (auto key : config_value.split(','))
+            table.set(key.to_lowercase());
+
+        return table;
+    };
+
+    Core::DirIterator dt(af_dir, Core::DirIterator::SkipDots);
+    while (dt.has_next()) {
+        auto af_name = dt.next_path();
+        auto af_path = String::format("%s/%s", af_dir.characters(), af_name.characters());
+        auto af = Core::ConfigFile::open(af_path);
+        if (!af->has_key("App", "Name") || !af->has_key("App", "Executable"))
```

Comment:
What if opening the config file fails?

Reply:
Core::ConfigFile can't fail - https://github.com/SerenityOS/serenity/blob/cd298446328af0d71a74d31f750fb1326689a3fe/Libraries/LibCore/ConfigFile.h#L42

The condition for checking name & executable means we will just ignore the file.

Reply:
We should teach that thing how to fail btw :)